spec/order_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "order"
+
+describe Order do
+
+  subject(:order) { described_class.new }
+
+  before do
+    allow(order).to receive(:send_sms)
```

Comment:
Watch out for this.  We're mocking the unit under test, which is dangerous, because it means we're not actually testing an accurate version of the real thing.  To avoid this, you could isolate the SMS dependency by injecting it.